tests/mocks.h

```diff
@@ -20,6 +23,22 @@ class MockBPFtrace : public BPFtrace {
   {
     return special_probes_;
   }
+
+  int resolve_uname(const std::string& name, struct symbol *sym, const std::string &path) const
```

Comment:
Not super important but defining this in mocks.cpp is probably better.

Also:
```suggestion
  int resolve_uname(const std::string& name, struct symbol *sym, const std::string &path) const override
```